lib/thrift/protocol/binary.ex

```diff
@@ -11,20 +11,21 @@ defmodule Thrift.Protocol.Binary do
   alias Thrift.TApplicationException
 
   @typedoc "Binary protocol field type identifier"
-  @type type_id :: (2..15)
+  @type type_id :: 0 | 2 | 3 | 4 | 6 | 8 | 10 | 11 | 12 | 13 | 14 | 15
```

Comment:
I like enumerating the specific values.

Zero doesn't belong in here though. A stop byte isn't a field type and isn't used in the same way. For example, it's not a valid argument to `skip_field/1`.